### DISCUSSION TITLE
Having a folder named "api" inside my UI root directory for API related frontend logic causes Next API routes to break

### DISCUSSION THREAD

**Discussion by agusterodin** [vercel/vercel#7048] [Help]
I have a folder called "api" in my UI project (Next.JS). My api folder contains logic related to interacting with my externally hosted non-serverless backend (using RTK query). These files are for the frontend codebase, not some sort of serverless function.

For example: 

![image](https://user-images.githubusercontent.com/10248395/142899338-b1831ede-0323-43d8-8978-f65888e49e44.png)

I do, however, want to use the api functionality inside Next. I use that for `next-auth`. When I go to deploy I am told

```
Warning: When using Next.js, it is recommended to place Node.js Serverless Functions inside of the `pages/api` (provided by Next.js) directory instead of `api` (provided by Vercel). Learn More: https://nextjs.org/docs/api-routes/introduction
```

The project compiles fine on the Vercel pipeline and I am left with a combination of both the files from the UI logic "api" folder as well as the "pages/api" folder contents as the output:

![image](https://user-images.githubusercontent.com/10248395/142899921-da471363-6509-4c7f-a25e-dc3592825714.png).

I would be fine just pretending that the files that got brought in weren't there as long as my catch-all route that `next-auth` uses didn't break. Unfortunately the catch-all route doesn't work and I get a 404

![image](https://user-images.githubusercontent.com/10248395/142900192-d542f220-cb17-46f6-a58b-ec90bf7cf483.png).

I tried creating a .vercelignore file containing the line `api` but then the Vercel pipeline doesn't see the "api" UI logic files at all, leading to a compilation error.

Why isn't there an option to disable serverless altogether? I understand that this is an edge case, but there is no reason that I shouldn't be able to put UI API logic inside of a folder called "api".

**Comment by agusterodin** [comment]
Hey, I noticed this message in my email notifications. Is there a link to the bug report @angusryer? Thanks

> I'm just submitting a bug report for this now.
> 
> Reproducible code: https://github.com/angusryer/missing-lambdas-vercel-nextjs
> Live deployment: https://missing-lambdas-vercel-nextjs.vercel.app/
> 
> It appears that when using NextJS /api routes, the contents of any folder (including /api itself) that contains an index.js/ts file will simple be replaced by a single api route lambda with the file contents of the index.js/ts. None of the folders other files will produce any build output. Oddly enough, the folder that contains the index file will remain--but without contents.
> 
> Ergo, avoid using index files in your routes for now.

**Reply by angusryer** [reply]
Hey yeah, sorry. I realized that my answer didn't actually answer your concern at all, so I removed it.

I submitted the bug report via one of Vercel's suggested pathways (https://github.com/vercel/vercel/issues/new/choose), and I think they process/post bugs themselves after review (? -- not sure, but I haven't seen my report posted in the issues log for the repo -- so I don't have link to it!)
